### PR TITLE
PoC of a process that display output only after a given time

### PR DESCRIPTION
**Pull request by sroze**
Follows up #38. 

Here's a simple PoC of an "interactive process API", that will display command output only if it's running since 5 seconds.

### CODE REVIEW

**Comment by tonypiper** [comment]
@sroze how good is this to go?

File: src/IO/Process/InteractiveProcess.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * @var WaitStrategy
+     */
+    private $waitStrategy;
+    /**
+     * @var Process
+     */
+    private $process;
+
+    public function __construct(Process $process, Pipe $pipe, WaitStrategy $waitStrategy)
```
Comment (sroze) [inline src/IO/Process/InteractiveProcess.php:29]: Missing PHPDocs

File: src/IO/Process/InteractiveProcess.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $this->waitStrategy = $waitStrategy;
+        $this->process = $process;
+    }
+
+    public function updatePipe(Pipe $pipe)
+    {
+        $this->pipe = $pipe;
+    }
+
+    public function updateWaitStrategy(WaitStrategy $waitStrategy)
```
Comment (sroze) [inline src/IO/Process/InteractiveProcess.php:47]: Can't be done at runtime so shouldn't be here. Using more the builder to setup these strategy would be better.

File: src/IO/Process/InteractiveProcessBuilder.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $this->userInteraction = $userInteraction;
+    }
+
+    public function getManagerFor($command)
+    {
+        return new InteractiveProcessManager(
+            new InteractiveProcess(
+                new Process($command),
+                new NullPipe(),
+                new BasicWait()
```
Comment (sroze) [inline src/IO/Process/InteractiveProcessBuilder.php]: See https://github.com/sroze/dock-cli/pull/39/files#r34266210

**Comment by sroze** [comment]
@tonypiper I prefer this way, could you review ?

**Comment by ciaranmcnulty** [comment]
This looks good to me

**Comment by sroze** [comment]
It shouldn't conflict a lot with #1, so merging.

**Pull request merged**